src/contexts/TelemetryContext.tsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable react-refresh/only-export-components */
+import { addParadata } from '@/api/07-paradata-events'
+import { useBatch } from '@/shared/hooks/useBatch'
+import type {
+  DefaultParadataValues,
+  TelemetryEvent,
+  TelemetryParadata,
+} from '@/types/telemetry'
+import {
+  createContext,
+  useCallback,
+  useContext,
+  useMemo,
+  useState,
+} from 'react'
+
+type TelemetryContextType = {
+  isTelemetryDisabled: boolean
+  pushEvent: (e: TelemetryParadata) => void | Promise<boolean>
+  setDefaultValues: (e: DefaultParadataValues) => void
+  triggerBatchTelemetryCallback?: () => Promise<void>
+}
+
+/** Mandatory values used as a context's last-resort fallback */
+const defaultValues = {
+  isTelemetryDisabled: true,
+  pushEvent: (_: TelemetryParadata) => {},
+  setDefaultValues: (_: DefaultParadataValues) => {},
+}
+
+/**
+ * Exposes shared functions to handle telemetry events.
+ *
+ * Should be used with the useBatch hook to reduce external API load.
+ */
+export const TelemetryContext: React.Context<TelemetryContextType> =
+  createContext(defaultValues)
+
+/**
+ * Returns the current telemetry context value.
+ *
+ * @version 1.2.4
+ * @see https://react.dev/reference/react/useContext
+ */
+export function useTelemetry() {
+  return useContext(TelemetryContext)
+}
+
+/** Initializes the telemetry context with a batch system */
+export function TelemetryProvider({
+  children,
+}: Readonly<{
+  children: React.ReactElement
+}>) {
+  const isTelemetryDisabled = import.meta.env.VITE_TELEMETRY_DISABLED === 'true'
```

Comment:
truc général : c'est préférable de mettre l'élément le plus constant à gauche de l'opérateur, c'est contre-intuitif pour notre sens de lecture mais c'est plus résilient (souvent) : 

```js
const isTelemetryDisabled = 'true' === import.meta.env.VITE_TELEMETRY_DISABLED
```

Quelques refs : 
- https://www.moserware.com/2008/01/constants-on-left-are-better-but-this.html
- https://softwareengineering.stackexchange.com/questions/74086/if-constant-variable-vs-if-variable-constant

Reply:
Le consensus n'est pas clair. Personnellement je trouve ça clairement plus lisible que le yoda style (et donc plus maintenable) mais on pourra en discuter et potentiellement lever une règle ESLint si l'équipe arrive à un consensus.

Reply:
Je suis d'accord avec Nicolas (traumatisme java null exception notamment ^^).

Reply:
On a discuté et Nicolas n'est plus d'accord avec lui-même.

Reply:
Oui en fait sur du js : avantage faible voire inexistant si on utilise un linter ou autre qui empêche les `if (x = 0)`, désavantage : ça rend le code un peu plus complexe à lire

=> ça vaut pas le coup